aptos-move/block-executor/src/scheduler_v2.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO(BlockSTMv2): enable dead code lint.
+#![allow(dead_code)]
+
+use aptos_infallible::Mutex;
+use aptos_mvhashmap::types::TxnIndex;
+use crossbeam::utils::CachePadded;
+use std::{
+    collections::BTreeSet,
+    sync::atomic::{AtomicU32, Ordering},
+};
+
+/// A structure that manages the execution queue for the BlockSTMv2 scheduler, and also
+/// exposes proxy interfaces to its implementation details (e.g. scheduler status).
+pub(crate) struct ExecutionQueueManager {
+    /// The first executed_idx many transctions have all finished their first incarnation,
+    /// i.e. have been executed at least once.
+    executed_idx: CachePadded<AtomicU32>,
+    /// Queue for scheduling transactions for execution.
+    /// TODO(BlockSTMv2): Alternative implementations for performance (e.g. packed ints,
+    /// intervals w. locks, CachePadded<ConcurrentQueue<TxnIndex>>).
+    execution_queue: Mutex<BTreeSet<TxnIndex>>,
+}
+
+impl ExecutionQueueManager {
+    // Note: is_first_reexecution must be determined and the method must be performed
+    // while holding the idx-th status lock.
+    pub(crate) fn add_to_schedule(&self, is_first_reexecution: bool, txn_idx: TxnIndex) {
+        // TODO(BlockSTMv2): Explain the logic for is_first_reexecution when SchedulerV2
+        // logic is added to the file.
+        if !is_first_reexecution || self.executed_idx.load(Ordering::Relaxed) >= txn_idx {
+            self.execution_queue.lock().insert(txn_idx);
```

Comment:
Add comment to when txn is add to the queue in the else case.

Reply:
Remind me what this was about?
The logic here is incomplete, and there is
        // TODO(BlockSTMv2): Explain the logic for is_first_reexecution when SchedulerV2 logic is added to the file.
is that also what you meant, or smt else?

Reply:
Explain where it is done. Ehi and when will add this txns to the queue.

Reply:
There is a TODO to add that explanation with the next PR, how can I explain it now when it's not here